File: leetcode/0066. Plus One/Solution.swift
// https://leetcode.com/problems/plus-one/

class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var result = [Int]()
        var carry = 1
        for digit in digits.reversed() {
            let sum = digit + carry
            result.append(sum % 10)
            carry = sum / 10
        }
        if carry > 0 {
            result.append(carry)
        }
        
        return result.reversed()
    }
}

File: leetcode/0066. Plus One/Solution.py
# https://leetcode.com/problems/plus-one/

from typing import List

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        result = []
        carry = 1
        for digit in reversed(digits):
            sum = digit + carry
            result.append(sum % 10)
            carry = sum // 10

        if carry > 0:
            result.append(carry)

        return list(reversed(result))

s = Solution()
assert s.plusOne([1,2,3]) == [1,2,4]
assert s.plusOne([4,3,2,1]) == [4,3,2,2]